ng/src/web/pages/groups/dialog.js

```diff
@@ -30,103 +31,109 @@ import Layout from '../../components/layout/layout.js';
 
 import PropTypes from '../../utils/proptypes.js';
 
-import withDialog from '../../components/dialog/withDialog.js';
+import SaveDialog from '../../components/dialog/savedialog.js';
 
 import Checkbox from '../../components/form/checkbox.js';
 import FormGroup from '../../components/form/formgroup.js';
 import MultiSelect from '../../components/form/multiselect.js';
 import TextField from '../../components/form/textfield.js';
 
+const DEFAULTS = {name: _('Unnamed'), users: []};
+
 const Dialog = ({
-    all_users,
-    comment,
+    allUsers,
     grant_full,
-    id,
-    name,
-    users,
-    onValueChange,
+    group,
+    title = _('New Group'),
+    visible,
+    onClose,
+    onSave,
   }) => {
+  const is_edit = is_defined(group);
 
-  const is_edit = is_defined(id);
   return (
-    <Layout flex="column">
-      <FormGroup title={_('Name')}>
-        <TextField
-          name="name"
-          grow="1"
-          value={name}
-          size="30"
-          onChange={onValueChange}
-          maxLength="80"/>
-      </FormGroup>
+    <SaveDialog
+      visible={visible}
+      title={title}
+      onClose={onClose}
+      onSave={onSave}
+      initialData={{
+        ...DEFAULTS,
+        ...group,
+      }}
+    >
+      {({
+        data: state,
+        onValueChange,
+      }) => {
+        const userOptions = map(allUsers, user => ({
```

Comment:
Should be moved up before the first jsx statement to avoid unnecessary re-creation too.